Automattic/jetpack
# Fix missing JS peer dependencies

#### Changes proposed in this Pull Request:

Mostly that means adding missing deps, but sometimes it's removing
unneeded deps that had peer deps. Then configure pnpm to start enforcing
peer deps.

root:
* Add eslint to pnpm.peerDependencyRules.ignoreMissing, we load eslint
  in js-tools but sometimes projects pull in extra eslint plugins on
  their own.
* Override @automattic/social-previews's dep on @wordpress/components
  from v12 to v19 to avoid a dep on @emotion/native that wants
  react-native. wp-calypso trunk has the same with no code changes.
* Delete @sveltejs/eslint-config's peer dep on eslint-plugin-node. It
  doesn't seem to actually make any use of it.

js-packages/licensing:
* @wordpress/components → react-dom

js-packages/shared-extension-utils
* @wordpress/compose → react

js-packages/storybook:
* @storybook/addon-essentials → babel-loader
* @storybook/react and others → react-dom, typescript
* postcss-loader → postcss

packages/my-jetpack:
* @storybook/testing-react → @storybook/addons, @storybook/client-api,
  @storybook/preview-web, @storybook/react → typescript
* @testing-library/react-hooks → react-test-renderer

plugins/boost:
* Remove unused deps: @wordpress/components, @wordpress/e2e-test-utils,
  @wordpress/element, prettier-plugin-svelte, sass, sass-loader

js-test-runner:
* @testing-library/react → react, react-dom
* @testing-library/react-hooks → react, react-test-renderer
* @wojtekmaj/enzyme-adapter-react-17 → react, react-dom
* @wordpress/jest-preset-default → react, react-dom

js-tools:
* @wordpress/eslint-plugin → typescript
* prettier-plugin-svelte → svelte

#### Jetpack product discussion


None

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Is CI happy?
* Try adding something without a peer dep or removing something that's needed as a peer dep. `pnpm install` should exit with a non-zero status.
